client/allocrunner/taskrunner/plugin_supervisor_hook.go

```diff
@@ -122,18 +141,46 @@ func (h *csiPluginSupervisorHook) Prestart(ctx context.Context,
 		return fmt.Errorf("failed to create mount point: %v", err)
 	}
 
+	if err := os.MkdirAll(h.socketMountPoint, 0700); err != nil && !os.IsExist(err) {
+		return fmt.Errorf("failed to create socket mount point: %v", err)
+	}
+
+	// where the socket will be mounted
 	configMount := &drivers.MountConfig{
 		TaskPath:        h.task.CSIPluginConfig.MountDir,
+		HostPath:        h.socketMountPoint,
+		Readonly:        false,
+		PropagationMode: "bidirectional",
+	}
+	// where the staging and per-alloc directories will be mounted
+	volumeStagingMounts := &drivers.MountConfig{
+		// TODO(tgross): add this TaskPath to the CSIPluginConfig as well
+		TaskPath:        "/local/csi",
```

Comment:
Note to reviewers: this PR is already huge and adding to this field to the jobspec is both a minor change and a lot of code to review. I'm going to do it in a follow-on PR.

Reply:
Not a very _good_ use case. 😁 

Suppose you had a plugin that required a configuration file that you created via `template` block, we could conceivably collide with the `destination` for that template. Ideally the user would just use a different destination of course. But obviously this is low-priority enough that it's not going to be a blocker at all for CSI GA.